db.js

```diff
@@ -54,8 +53,16 @@ var books = (function () {
 var categories = (function () {
   return {
     get: function (categoryId) {
-      return pool
-        .query("SELECT * FROM categories WHERE id = ?", [categoryId])
+      return pool.query("SELECT * FROM categories WHERE id = ?", [categoryId])
+        .then(([rows, fields]) => {
+          if (!rows || rows.length === 0) {
+            return null;
+          }
+          return rows[0];
+        });
+    },
+    getAllByUserId: function (category) {
+      return pool.query("SELECT * FROM categories WHERE user_id = ?", [category.user_id])
```

Comment:
This is fine for now, but it may be overkill to pass in a whole category object. You could probably get away with just passing the userId